scripts/seed-projects/seed-projects.py

```diff
@@ -0,0 +1,101 @@
+import argparse, json, requests
+from faker import Faker
+from random import randint
+
+def create_fake_publication(fake):
+    return {
+        "doi": fake.isbn13(), # Not a real doi
+        "url": fake.url(),
+        "journalTitle": fake.sentence(3),
+        "title": fake.sentence(15),
+        "authors": [ fake.name() for i in range(8) ]
+    }
+
+def create_fake_contributor(fake):
+    return {
+        "name": fake.name(),
+        "email": fake.email(),
+        "institution": fake.company(),
+        "corresponding_contributor": False
+    }
+
+def create_project(fake, catalogued, ingest_url, token):
+    project = {
+        "content": {
+            "contributors": [ create_fake_contributor(fake) for i in range(randint(1,10)) ],
+            "describedBy": "https://schema.dev.data.humancellatlas.org/type/project/14.3.0/project",
+            "funders": [{
+                "grant_id": fake.sentence(4),
+                "organization": fake.company()
+            }],
+            "project_core": {
+                "project_short_name": fake.password(10),
+                "project_title": fake.sentence(10),
+                "project_description": fake.paragraph(4)
+            },
+            "publications": [{
+                "authors": [ fake.name() for i in range(8) ],
+                "title": fake.sentence(15)
+            }],
+            "schema_type": "project"
+        },
+        "isInCatalogue": catalogued,
+        "organ": {
+            "ontologies": [{
+                "text": "future glans",
+                "ontology": "UBERON:0013238",
+                "ontology_label": "future glans"
+            }]
+        },
+        "technology": {
+            "ontologies": [{
+                "text": "sci-RNA-seq",
+                "ontology": "EFO:0010550",
+                "ontology_label": "sci-RNA-seq"
+            }]
+        },
+        "identifyingOrganisms": ["Human"],
+        "releaseDate": "2024-03-10T23:00:00.000Z",
+        "dataAccess": {
+            "type": "All fully open"
+        },
+        "publicationsInfo": [ create_fake_publication(fake) for i in range(randint(1,3)) ]
+    }
+
+    try:
+        url = f'{ingest_url}/projects'
+        headers={'Authorization': token, 'Content-Type': 'application/json'}
+        data = json.dumps(project)
+        r = requests.post(url, data=data, headers=headers)
```

Comment:
you can specify json param and you won't actually need to pass the Content-Type header

requests.post(url, json=data, headers=headers)

Reply:
Cool didn't know that! You learn something new everyday